Sensors.cpp

```diff
@@ -471,6 +471,154 @@ uint8_t Baro_update() {                   // first UT conversion is started in i
 }
 #endif
 
+// ************************************************************************************************************
+// I2C Barometer BOSCH BMP280
+// ************************************************************************************************************
+// I2C adress: 0x76 or 0x77 !!!Please make sure you set the correct address (BMP280_ADDR)
+// ************************************************************************************************************
+
+#if defined(BMP280)
+
+#define BMP280_ADDR  0x76 
+#define i2c_read(ack)  (ack) ? i2c_readAck() : i2c_readNak(); 
+
+#define BMP280_CAL_REG_FIRST  0x88
+#define BMP280_CAL_REG_LAST 0xA1
+#define BMP280_CAL_DATA_SIZE  (BMP280_CAL_REG_LAST+1 - BMP280_CAL_REG_FIRST)
+
+#define BMP280_STATUS_REG 0xF3
+#define BMP280_CONTROL_REG  0xF4
+#define BMP280_CONFIG_REG 0xF5
+
+#define BMP280_PRES_REG   0xF7
+#define BMP280_TEMP_REG   0xFA
+#define BMP280_RAWDATA_BYTES  6 // 3 bytes pressure, 3 bytes temperature
+
+int64_t p; //temp variable for calculating pressure  
+uint32_t deadline;
+
+static union _bmp280_cal_union {
+  uint8_t bytes[BMP280_CAL_DATA_SIZE];
+  struct {
+    uint16_t dig_t1;
+    int16_t  dig_t2;
+    int16_t  dig_t3;
+    uint16_t dig_p1;
+    int16_t  dig_p2;
+    int16_t  dig_p3;
+    int16_t  dig_p4;
+    int16_t  dig_p5;
+    int16_t  dig_p6;
+    int16_t  dig_p7;
+    int16_t  dig_p8;
+    int16_t  dig_p9;
+  };
+} bmp280_cal;
+
+/*
+ * read calibration registers
+ */
+static void bmp280_getcalibration(void)
+{
+  memset(bmp280_cal.bytes, 0, sizeof(bmp280_cal));
+
+  i2c_read_reg_to_buf(BMP280_ADDR, 
+    BMP280_CAL_REG_FIRST,
+    bmp280_cal.bytes,
+    BMP280_CAL_DATA_SIZE
+  );
+}
+
+void bmp280_set_ctrl(uint8_t osrs_t, uint8_t osrs_p, uint8_t mode)
+{
+  i2c_writeReg(BMP280_ADDR, BMP280_CONTROL_REG,
+    ((osrs_t & 0x7) << 5) | ((osrs_p & 0x7) << 2) | (mode & 0x3));
+}
+
+void bmp280_set_config(uint8_t t_sb, uint8_t filter, uint8_t spi3w_en)
+{
+  i2c_writeReg(BMP280_ADDR, BMP280_CONFIG_REG,
+    ((t_sb & 0x7) << 5) | ((filter & 0x7) << 2) | (spi3w_en & 1));
+}
+
+#define bmp280_24bit_reg(b1, b2, b3)  ( \
+  ((int32_t)(b1) << 16) \
+  | ((int32_t)(b2) << 8) \
+  | ((int32_t)(b3) ) \
+)
+
+/*  Measures and calculates pressure and temperature
+ *   
+ *  This function updates global variables baroPressure and baroTemperature
+ *  
+ *  baroTemperature unit is 0.01 deg C 
+ *  baroPressure unit is Pa  
+ *   
+ */
+ 
+void bmp280_measure(void)
+{
+  uint8_t data[BMP280_RAWDATA_BYTES];
+  int32_t temp_raw, pres_raw, var1, var2, t_fine;
```

Comment:
Variables var1 and var2 must be int64 type.

Reply:
I could not find anything can you please help